### PR TITLE
Enhancement/skale 3272 repair mode

### PR DESCRIPTION
**Pull request by badrogger**

### CODE REVIEW

File: core/schains/cleaner.py [hunk 1]
```diff
@@ -162,9 +167,13 @@ def delete_bls_keys(skale, schain_name):
...
-            continue
+            if os.path.isfile(secret_key_share_filepath):
+                secret_key_share_config = read_json(secret_key_share_filepath) or {}
+                bls_key_name = secret_key_share_config.get('key_share_name')
+                if bls_key_name:
+                    sgx = SgxClient(SGX_SERVER_URL,
+                                    path_to_cert=SGX_CERTIFICATES_FOLDER)
+                    sgx.delete_bls_key(bls_key_name)
+        except Exception as err:
+            logger.warning(f'Removing secret_key for rotation {i} failed',
```
Comment (dmytrotkk) [inline core/schains/cleaner.py]: why `logger.warning` and not `logger.exception`?

**Review by dmytrotkk — CHANGES_REQUESTED** [review]

File: core/schains/cleaner.py [hunk 2]
```diff
@@ -56,27 +57,30 @@ def run_cleaner(skale, node_config):
 
 
 def log_remove(component_name, schain_name):
-    logger.warning(f'Going to remove {component_name} for sChain {schain_name}')
+    logger.info(f'Going to remove {component_name} for sChain {schain_name}')
 
 
-def remove_schain_volume(schain_name):
+def remove_schain_volume(schain_name, dutils=None):
```
Comment (DmytroNazarenko) [inline core/schains/cleaner.py]: Mb add type of params like in remove_schain_container

File: core/schains/cleaner.py (hunk 1 above)
Comment (badrogger) [reply core/schains/cleaner.py]: addressed

**Review by dmytrotkk — DISMISSED** [review]
Great!

File: core/schains/creator.py [hunk 3]
```diff
@@ -179,41 +209,33 @@ def monitor_schain(skale, node_info, schain, ecdsa_sgx_key_name):
...
-        schain=schain,
-        checks=checks_dict,
-        node_id=node_id,
-        sgx_key_name=sgx_key_name,
-        rotation=rotation,
-        ecdsa_sgx_key_name=ecdsa_sgx_key_name,
-        schain_record=schain_record
-    )
-
+        logger.info(f'Running monitor for sChain {name} in REGULAR mode')
```
Comment (DmytroNazarenko) [inline core/schains/creator.py]: Let's unify logger.info(f'Running monitor for sChain {name} in {mode} mode')

File: core/schains/volume.py [hunk 4]
```diff
@@ -74,5 +76,5 @@ def get_filestorage_info():
     return read_json(FILESTORAGE_ARTIFACTS_FILEPATH)
 
 
-def get_schain_volume_config(name, mount_path):
-    return {f'{name}': {'bind': mount_path, 'mode': 'rw'}}
+def get_schain_volume_config(name, mount_path, mode='Z'):
+    return {f'{name}': {'bind': mount_path, 'mode': 'Z'}}
```
Comment (DmytroNazarenko) [inline core/schains/volume.py]: 'mode': 'Z' -> 'mode': mode

File: scripts/run_tests.sh [hunk 5]
```diff
@@ -31,11 +31,11 @@ bash scripts/run_redis.sh
...
-py.test tests/rotation_test
+# py.test tests/schains/creator_test.py --ignore=tests/firewall --ignore=tests/rotation_test
+# export SGX_CERTIFICATES_FOLDER=$PWD/tests/skale-data/node_data/sgx_certs
+# mkdir -p $SGX_CERTIFICATES_FOLDER
+# rm -rf $SGX_CERTIFICATES_FOLDER/sgx.*
+# py.test tests/rotation_test/
 find . -name \*.pyc -delete
 scripts/run_firewall_test.sh
-rm -r $SGX_CERTIFICATES_FOLDER
+# rm -r $SGX_CERTIFICATES_FOLDER
```
Comment (DmytroNazarenko) [inline scripts/run_tests.sh]: Uncomment or remove

File: tests/conftest.py [hunk 6]
```diff
@@ -1,122 +1,191 @@
...
+    "key_share_name": "BLS_KEY:SCHAIN_ID:33333333333333333333333333333333333333333333333333333333333333333333333333333:NODE_ID:0:DKG_ID:0"  # noqa
 }
 
 
 @pytest.fixture
-def schain_dir():
-    schain_dir_path = os.path.join(SCHAINS_DIR_PATH, SCHAIN_NAME)
+def _schain_name():
+    """ Generates default schain name """
+    return 'schain_' + get_random_string()
```
Comment (DmytroNazarenko) [inline tests/conftest.py]: schain_name should be without 'schain' prefix

File: tests/firewall/run_test.py [hunk 7]
```diff
@@ -79,7 +77,7 @@ def test_rules_manipulation():
         assert plain_rule not in plain_rules
 
 
-@pytest.mark.skip('Problems with github actions')
+# @pytest.mark.skip('Problems with github actions')
```
Comment (DmytroNazarenko) [inline tests/firewall/run_test.py]: Mb remove?

**Review by DmytroNazarenko — CHANGES_REQUESTED** [review]

File: tests/firewall/run_test.py (hunk 7 above)
Comment (badrogger) [reply tests/firewall/run_test.py]: addressed

File: tests/conftest.py (hunk 6 above)
Comment (badrogger) [reply tests/conftest.py]: addressed

File: scripts/run_tests.sh (hunk 5 above)
Comment (badrogger) [reply scripts/run_tests.sh]: addressed

File: core/schains/volume.py (hunk 4 above)
Comment (badrogger) [reply core/schains/volume.py]: addressed

File: core/schains/creator.py (hunk 3 above)
Comment (badrogger) [reply core/schains/creator.py]: addressed

File: core/schains/cleaner.py (hunk 2 above)
Comment (badrogger) [reply core/schains/cleaner.py]: addressed

File: core/schains/creator.py [hunk 8]
```diff
@@ -112,42 +126,58 @@ def monitor(skale, node_config):
...
-
     if not schain_record.first_run:
         notify_checks(name, node_info, checks_dict)
 
-    first_run = schain_record.first_run
     schain_record.set_first_run(False)
-    if rotation_in_progress and exiting_node:
-        logger.warning(f'Node is exiting. sChain {name} will be stoped at {finish_time}')
+    logger.info(f'Running monitor for sChain {name} in {mode.name} mode. '
+                f'Finish time: {finish_time}')
```
Comment (DmytroNazarenko) [inline core/schains/creator.py]: Finish time log doesn't  need for REGULAR MonitorNode

**Review by DmytroNazarenko — CHANGES_REQUESTED** [review]

File: core/schains/creator.py (hunk 8 above)
Comment (badrogger) [reply core/schains/creator.py]: addressed

**Review by DmytroNazarenko — APPROVED** [review]

**Review by dmytrotkk — APPROVED** [review]

**Pull request merged**